datafusion-cli/src/print_format.rs

```diff
@@ -161,23 +161,29 @@ impl PrintFormat {
         maxrows: MaxRows,
         with_header: bool,
     ) -> Result<()> {
-        if batches.is_empty() || batches[0].num_rows() == 0 {
+        // filter out any empty batches
+        let batches: Vec<_> = batches
+            .iter()
+            .filter(|b| b.num_rows() > 0)
+            .cloned()
+            .collect();
+        if batches.is_empty() {
```

Comment:
What will happen for queries that return empty results?

e.g.

```sql
select * from table where 1=0
```

Will this print nothing at all, not even the table outline?

Reply:
I would also like to mention that a similar behavior exists for streaming results. If the first batch of the stream is empty, the header will not be printed, even if it was set to true (this issue existed prior to this PR). Perhaps we should print an empty record batch for the header before these checks if the flag is set.

Reply:
> Will this print nothing at all, not even the table outline?

Right, it will NOT print the outline / header. Also see https://github.com/apache/arrow-datafusion/pull/8895#discussion_r1455440663 , i.e. I also find this rather confusing. However this PR here just fixes the status quo to not swallow the output for aggregates. I personally think we should change the behavior in a follow-up to at least print the table outline / header even for empty results.

Reply:
Ah ok, I didn't realize the "print nothing if no data" was a previous issue before that streaming PR

Then yeah I agree can fix that separately if it wasn't technically a regression

Reply:
Filed https://github.com/apache/arrow-datafusion/issues/8904 to track (and I confirmed the "print nothing if no data" behavior was present in versions 33 and 25).